libid/complex.cpp

```diff
@@ -0,0 +1,536 @@
+// SPDX-License-Identifier: GPL-3.0-only
+//
+// Complex.cpp: interface for the Complex class.
+//
+//////////////////////////////////////////////////////////////////////
+#include "Complex.h"
+
+//Complex::Complex(void)
+//    {
+//    }
+
+Complex::~Complex(void)
```

Comment:
In C++, there's never any need to use `(void)` as an argument list like C requires.  Just write `()`.

Furthermore, since this destructor doesn't do anything, declare that the compiler should explicitly provide a destructor by writing in the class declaration (in the header):
```
class Complex
{
public:
    ~Complex() = default;
...
};
```